modules/build/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -293,12 +296,11 @@ final case class BuildOptions(
     val sortedExtraPlatforms = scalaOptions0
       .extraPlatforms
       .toVector
-      .sorted
-    this +: sortedExtraPlatforms.map { pf =>
+    this +: sortedExtraPlatforms.map { case (pf, pos) =>
       copy(
         scalaOptions = scalaOptions0.copy(
-          platform = Some(pf),
-          extraPlatforms = Set.empty
+          platform = Some(Positioned(List(pos.positions.head), pf)),
```

Comment:
Why not this? (it doesn't discard positions this way)
```suggestion
          platform = Some(Positioned(pos.positions, pf)),
```